src/java.desktop/share/classes/javax/swing/JFileChooser.java

```diff
@@ -277,13 +278,13 @@
 
     private boolean dragEnabled = false;
 
-    private FileFilter fileFilter = null;
+    private @Nullable FileFilter fileFilter = null;
 
-    private FileSystemView fileSystemView = null;
+    private @Nullable FileSystemView fileSystemView = null;
 
-    private File currentDirectory = null;
-    private File selectedFile = null;
-    private File[] selectedFiles;
+    private @Nullable File currentDirectory = null;
+    private @Nullable File selectedFile = null;
+    private @Nullable File[] selectedFiles;
```

Comment:
Based on the code, I believe this should be `File @Nullable []` (a possibly-null array of non-null files) rather than `@Nullable File[]` (a non-null array of possibly-null files).

Reply:
Agreed, pending the comment above about possibly allowing nulls inside the array.